ckanext/adfs/plugin.py

```diff
@@ -17,14 +18,9 @@
 X509 = ''
 WSFED_ENDPOINT = ''
 WTREALM = pylons.config['adfs_wtrealm']
-with open(pylons.config['adfs_federation_metadata_path']) as federation_meta:
-    METADATA = ET.fromstring(federation_meta.read())
-    for child in list(METADATA):
-        service_type = child.attrib.get('{http://www.w3.org/2001/XMLSchema-instance}type', None)
-        if service_type == 'fed:ApplicationServiceType':
-            X509 = child.find('{urn:oasis:names:tc:SAML:2.0:metadata}KeyDescriptor').xpath('string()')
-            WSFED_ENDPOINT = child.find('{http://docs.oasis-open.org/wsfed/federation/200706}PassiveRequestorEndpoint').xpath('string()')
-            break
+METADATA = get_federation_metadata(pylons.config['adfs_federation_metadata_url'])
```

Comment:
Yeah... that's a conscious decision on my part (realising we had the conversation about this yesterday). Pinging out to M$ every time there's a login just feels inelegant and clunky to me - at least when I was writing the code yesterday I was asking myself, "hang on, surely there's a better way than this?" and made a judgement call there and then. My train of thought was simply that if the cert changes we only need to restart the apache processes and the new settings are automatically loaded. Happy to discuss (but upon reflection I think this is the best solution given a broken situation). ;-)